core/src/data_sources/data_source.rs

```diff
@@ -1166,37 +1166,53 @@ impl DataSource {
                 let data_source = data_source.clone();
 
                 tokio::spawn(async move {
-                    let mut d: Document = match store
+                    match store
                         .load_data_source_document(&project, &data_source_id, &document_id, &None)
                         .await?
                     {
-                        Some(d) => d,
-                        None => Err(anyhow!("Document not found"))?,
-                    };
-
-                    if full_text {
-                        let document_id_hash = make_document_id_hash(&document_id);
-
-                        let stored_doc = FileStorageDocument::get_stored_document(
-                            &data_source,
-                            d.created,
-                            &document_id_hash,
-                            &d.hash,
-                        )
-                        .await?;
-
-                        d.text = Some(stored_doc.full_text.clone());
+                        Some(mut d) => {
+                            if full_text {
+                                let document_id_hash = make_document_id_hash(&document_id);
+
+                                let stored_doc = FileStorageDocument::get_stored_document(
+                                    &data_source,
+                                    d.created,
+                                    &document_id_hash,
+                                    &d.hash,
+                                )
+                                .await?;
+
+                                d.text = Some(stored_doc.full_text.clone());
+                            }
+                            Ok::<Option<Document>, anyhow::Error>(Some(d))
+                        }
+                        None => {
+                            // document not found should never happen
```

Comment:
Let's try to go all the ways to column 100 for comments :+1:

Reply:
Done in https://github.com/dust-tt/dust/pull/7957